src/Core/Table/Factory/TableFactory.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core\Table\Factory;
+
+use PrestaShop\PrestaShop\Core\Table\Definition\TableDefinitionInterface;
+use PrestaShop\PrestaShop\Core\Table\Table;
+use PrestaShopBundle\Service\Hook\HookDispatcher;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class TableFactory is responsible for creating table from it's definition
+ */
+final class TableFactory implements TableFactoryInterface
+{
+    /**
+     * @var HookDispatcher
+     */
+    private $dispatcher;
+
+    /**
+     * @var FormFactoryInterface
+     */
+    private $formFactory;
+
+    /**
+     * @param HookDispatcher $dispatcher
+     * @param FormFactoryInterface $formFactory
+     */
+    public function __construct(
+        HookDispatcher $dispatcher,
+        FormFactoryInterface $formFactory
+    ) {
+        $this->dispatcher = $dispatcher;
+        $this->formFactory = $formFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createFromDefinition(TableDefinitionInterface $tableDefinition, Request $request) {
+        // Execute hook to allow developers to modify/extend table definition
+        // For exmaple add new columns, row actions & etc.
+        $this->dispatcher->dispatchForParameters('modifyTableDefinition', [
```

Comment:
shouldn't be unique event for each "table"? How can you add/remove/alter a column of - let's says - Logs pages theorically?

Reply:
each $tableDefinition has getIdentifier() which can be 'request_sql_table', so check can be performed in hook, but dynamic hooks is also an option here.

Reply:
ok then, (in all cases) we will document everything. And when I say everything I mean a PrestaShop developer should have information to:
* add/remove or alter columns
* re-order columns
* change view of a specific column case (Think the specific view of Employee column in Logs page for instance)

If we have all these things, do we have all things we need compared to 1.6.1.x?